site/source/docs/tools_reference/emcc.rst

```diff
@@ -282,6 +282,9 @@ Options that are modified or new in *emcc* are listed below:
 ``--exclude-file <name>``
 	Files and directories to be excluded from :ref:`--embed-file <emcc-embed-file>` and :ref:`--preload-file <emcc-preload-file>`. Wildcards (*) are supported.
 
+``--fully-prepare-files``
```

Comment:
The name does not somehow feel good, both terms "fully" and "prepare" sound a bit odd and generic/vague. If this option is not set, the files could be thought to be "unprepared", but that has an odd ring, I'm not sure I like the concept of "preparing" files. If this fully prepares the files, then is it possible to have the files partially prepared in some other way(?)

However I'm not sure that I can come up with any better name. The most descriptive names to me are `--decode-assets-at-startup` or `--decode-for-synchronous-access`, which are a bit verbose. Perhaps `--auto-decode-assets`?

Not sure, I'm ok whichever way, the feature looks solid anyways. Thanks for working on this, I had mentally put this on my work queue, so happy to see you got to it first! :)